mainwindow.cc

```diff
@@ -795,6 +795,13 @@ MainWindow::MainWindow( Config::Class & cfg_ ):
   {
     // Upgrade existing dummy tray icon into a full-functional one
 
+#if QT_VERSION >= 0x050000
+#ifdef __linux__
+    trayIcon->deleteLater();
+    trayIcon = new QSystemTrayIcon( QIcon::fromTheme("goldendict-tray", QIcon( ":/icons/programicon_old.png" )), this );
```

Comment:
I looked at the `updateTrayIcon()` implementation and it seems to me that you can optimize the code for this case:
```cpp
#if ...
    trayIcon->deleteLater();
    trayIcon = NULL;
#else
    trayIcon->setContextMenu( &trayIconMenu );
    trayIcon->show();

    connect( trayIcon, SIGNAL( activated( QSystemTrayIcon::ActivationReason ) ),
this, SLOT( trayIconActivated( QSystemTrayIcon::ActivationReason ) ) );
#endif
```

Reply:
I think that `delete trayIcon` would be better than [`QObject::deleteLater()`](https://doc.qt.io/qt-5/qobject.html#deleteLater) here because this is the `MainWindow` constructor, which is called directly from `main()` and so there is no event loop.